src/app/beer-detail/beer-detail.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Location } from '@angular/common';
+import { Beer } from '../beer';
+import { BeerService } from '../beer.service';
+
+@Component({
+    templateUrl: './beer-detail.component.html',
+    styleUrls: ['./beer-detail.component.css']
+})
+export class BeerDetailComponent implements OnInit {
+    @Input() beer: Beer;
+
+    constructor(
+        private route: ActivatedRoute,
+        private beerService: BeerService,
+        private location: Location
+    ) { }
+
+    ngOnInit() {
+        this.getBeer();
+    }
+    private getBeer(): void {
+        const id = +this.route.snapshot.paramMap.get('id');
```

Comment:
this works, but will not work if you try to switch from one detail to another just changing the route id
it's better to use an observable: `this.route.paramMap.subscribe((params) => {})`